persistence/pg.go

```diff
@@ -301,6 +287,28 @@ func (p *PostgresPersister) MarkHtlcSettled(ctx context.Context,
 	return invoiceSettled, nil
 }
 
+func (p *PostgresPersister) selectInvoiceForUpdate(ctx context.Context,
+	tx *pg.Tx, key types.HtlcKey) (*dbInvoice, error) {
+
+	// Look up the htlc hash.
+	hash, err := p.getHtlcHash(ctx, tx, key)
+	if err != nil {
+		return nil, err
+	}
+
+	var invoice dbInvoice
+	err = tx.ModelContext(ctx, &invoice).
+		Where("hash=?", hash).
+		For("UPDATE").
+		Select()
+	switch {
```

Comment:
This can be an if statement?

Or you are missing the `err != nil` case?

Reply:
It was not a bug because it was:

```
var invoice Invoice

err := SelectForUpdate(invoice)
switch{
       case err == pg.NoRows....
}

return invoice, err

```

So if err !=nil, the error was returned and check by the caller

But it was not very clean..